src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -487,9 +488,13 @@ public bool WriteIndented
             {
                 VerifyMutable();
                 _referenceHandler = value;
+                ReferenceHandlingStrategy = value?.HandlingStrategy ?? ReferenceHandlingStrategy.None;
```

Comment:
nit (this didn't change in this PR), but why do we allow `null` to be passed as input to this setter?

Reply:
it is the default value of this property and it means "no handling", it is also similar to `JsonPropertyNamingPolicy` and `Encoder` allowing `null`.